test/e2e/e2e_test.go

```diff
@@ -259,6 +259,123 @@ var _ = ginkgo.Describe("Kueue", func() {
+			lookupKey1 := types.NamespacedName{Name: sampleJob.Name, Namespace: sampleJob.Namespace}
+			//createdJob1 := &batchv1.Job{}
+			wll := &kueue.Workload{}
+
+			ginkgo.By("checking the job starts")
+			gomega.Expect(k8sClient.Create(ctx, sampleJob)).Should(gomega.Succeed())
+
+			lookupKey := types.NamespacedName{Name: sampleJob.Name, Namespace: sampleJob.Namespace}
+			createdJob := &batchv1.Job{}
+			gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob)).Should(gomega.Succeed())
+
+			gomega.Eventually(func() *bool {
+				gomega.Expect(k8sClient.Get(ctx, lookupKey1, sampleJob)).Should(gomega.Succeed())
+				return sampleJob.Spec.Suspend
+			}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(false)))
+
+			wlKey := types.NamespacedName{Name: workloadjob.GetWorkloadNameForJob(sampleJob.Name), Namespace: sampleJob.Namespace}
+
+			gomega.Eventually(func() bool {
+				if err := k8sClient.Get(ctx, wlKey, wll); err != nil {
+					return false
+				}
+				return workload.HasQuotaReservation(wll)
+			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+
+			ginkgo.By("Change the QueueingPolicy, suspend job and check the job remains suspended and the workload evicted")
+			// Changing QueuingPolicy to Never
+			wll.Spec.QueueingPolicy = kueue.QueueingPolicyTypeNever
+			gomega.Expect(k8sClient.Update(ctx, wll)).Should(gomega.Succeed())
+
+			// Suspending the job
+			gomega.Expect(k8sClient.Get(ctx, lookupKey, createdJob)).Should(gomega.Succeed())
+			createdJob.Spec.Suspend = ptr.To(true)
+			gomega.Expect(k8sClient.Update(ctx, createdJob)).Should(gomega.Succeed())
+
+			// Checking job stays suspended
+			ginkgo.By("checking job stays suspended")
+			gomega.Eventually(func() *bool {
+				gomega.Expect(k8sClient.Get(ctx, types.NamespacedName{Name: sampleJob.Name, Namespace: sampleJob.Namespace}, createdJob)).
+					Should(gomega.Succeed())
+				return createdJob.Spec.Suspend
+			}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(true)))
+
+			// Workload should get unadmitted
+			gomega.Expect(k8sClient.Get(ctx, wlKey, wll)).Should(gomega.Succeed())
+			util.ExpectWorkloadsToBePending(ctx, k8sClient, wll)
+
+			// Workload should stay pending
+			gomega.Consistently(func() bool {
+				if err := k8sClient.Get(ctx, client.ObjectKeyFromObject(wll), wll); err != nil {
+					return false
+				}
+				return workload.HasQuotaReservation(wll)
+			}, util.ConsistentDuration, util.Interval).Should(gomega.BeFalse())
+
+			util.FinishEvictionForWorkloads(ctx, k8sClient, wll)
+
+			ginkgo.By("checking the jobs becomes unsuspended after we update the QueueingPolicy back to Always")
+			gomega.Eventually(func() kueue.QueueingPolicyType {
+				gomega.Expect(k8sClient.Get(ctx, wlKey, wll)).Should(gomega.Succeed())
+				wll.Spec.QueueingPolicy = kueue.QueueingPolicyTypeAlways
+				gomega.Expect(k8sClient.Update(ctx, wll)).Should(gomega.Succeed())
```

Comment:
```suggestion
				return k8sClient.Update(ctx, wll)).Should(gomega.Succeed()
```

and check that it succeeds